Documentation/DesignDocs/corewcfqueue-design.md

```diff
@@ -0,0 +1,190 @@
+# Objective 
+The objective of this document is to provide details on how to build a generic queue transport inside CoreWCF so that it can enable various queue binding. This document is based on the discussion outlined here https://github.com/CoreWCF/CoreWCF/issues/164 .
+
+# Why are we building
+When WCF was build there were not many queing technique and there was no concept of cloud. With multiple cloud providers supporting queing technique and many queing technique widely available, we wanted to build a generic transport, so that various queuing mechanism can be supported in CoreWCF.
```

Comment:
Spelling/grammar/phrasing:  
When WCF was built there were not many queuing technologies....

Reply:
Those same changes should be made through the entire paragraph and not just the one sentence.